pkg/clustermesh/mcsapi/cell.go

```diff
@@ -113,5 +115,27 @@ func initMCSAPIController(params mcsAPIParams) error {
 	}
 
 	params.Logger.Info("Multi-Cluster Services API support enabled")
+
+	remoteClusterServiceSource := &remoteClusterServiceSource{Logger: params.Logger}
+	params.ClusterMesh.RegisterClusterServiceExportUpdateHook(remoteClusterServiceSource.onClusterServiceExportUpdate)
+	params.ClusterMesh.RegisterClusterServiceExportDeleteHook(remoteClusterServiceSource.onClusterServiceExportUpdate)
+	svcImportReconciler := newMCSAPIServiceImportReconciler(
+		params.CtrlRuntimeManager, params.Logger, params.ClusterInfo.Name,
+		params.ClusterMesh.GlobalServiceExports(), remoteClusterServiceSource,
+	)
+
+	params.JobGroup.Add(job.OneShot("mcsapi-main", func(ctx context.Context, health cell.Health) error {
+		params.Logger.Info("Bootstrap Multi-Cluster Services API support")
+
+		if err := params.ClusterMesh.ServiceExportsSynced(ctx); err != nil {
+			return nil // The parent context expired, and we are already terminating
+		}
+
+		if err := svcImportReconciler.SetupWithManager(params.CtrlRuntimeManager); err != nil {
```

Comment:
OneShot jobs should terminate when the supplied `ctx` is closed. I don't see `ctx` being passed to `SetupWithManager()`; unless there's something I'm missing, there's nothing that will cleanly stop this controller?

Reply:
Hmm I am not sure how controller-runtime controllers handle their shutdown TBH. From a quick look I haven't been able to find some other cilium controller-runtime controllers that do something differently from this :/.

FYI the one shot job is to not make sure the `ServiceExportsSynced` doesn't hold the start of other hive components in the operator.

Reply:
aha, I missed that this was a shared controller-runtime manager! In which case, disregard this :-)